src/SysInfo.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 - Ed Beroset <beroset@ieee.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef SYSINFO_H
+#define SYSINFO_H
+
+#include <sys/sysinfo.h>
+
+#include <QObject>
+#include <QtQml>
+
+class SysInfo : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(long uptime READ uptime NOTIFY uptimeChanged)
+    Q_PROPERTY(QList<qreal> loads READ loads NOTIFY loadsChanged)
+    Q_PROPERTY(unsigned long totalram READ totalram NOTIFY totalramChanged)
+    Q_PROPERTY(unsigned long freeram READ freeram NOTIFY freeramChanged)
+    Q_PROPERTY(unsigned long sharedram READ sharedram NOTIFY sharedramChanged)
+    Q_PROPERTY(unsigned long bufferram READ bufferram NOTIFY bufferramChanged)
+    Q_PROPERTY(unsigned long totalswap READ totalswap NOTIFY totalswapChanged)
+    Q_PROPERTY(unsigned long freeswap READ freeswap NOTIFY freeswapChanged)
+    Q_PROPERTY(unsigned procs READ procs NOTIFY procsChanged)
+    Q_PROPERTY(unsigned long totalhigh READ totalhigh NOTIFY totalhighChanged)
+    Q_PROPERTY(unsigned long freehigh READ freehigh NOTIFY freehighChanged)
+    Q_PROPERTY(unsigned mem_unit READ mem_unit CONSTANT)
+public:
+    SysInfo(QObject *parent = nullptr);
+    unsigned long uptime() const { return info.uptime; }
+    QList<qreal> loads() const { return m_loads; }
+    unsigned long totalram() const { return info.totalram; }
+    unsigned long freeram() const { return info.freeram; }
+    unsigned long sharedram() const { return info.sharedram; }
+    unsigned long bufferram() const { return info.bufferram; }
+    unsigned long totalswap() const { return info.totalswap; }
+    unsigned long freeswap() const { return info.freeswap; }
+    unsigned short procs() const { return info.procs; }
+    unsigned long totalhigh() const { return info.totalhigh; }
+    unsigned long freehigh() const { return info.freehigh; }
+    unsigned mem_unit() const { return info.mem_unit; }
+
+signals: 
+    void uptimeChanged();
+    void loadsChanged();
+    void totalramChanged();
+    void freeramChanged();
+    void sharedramChanged();
+    void bufferramChanged();
+    void totalswapChanged();
+    void freeswapChanged();
+    void procsChanged();
+    void totalhighChanged();
+    void freehighChanged();
+
+public slots:
+    void refresh() {
```

Comment:
Is there a reason why you prefer having the code of the function in the header ? The informal convention I've noticed is that occasional trivial one liners sometimes stay in the headers but generally speaking every implementation tends to be in the .cpp, and definitely if they are long functions. (Otherwise the reader spends quite some time looking for it)

One justification for this practice is that, when a compilation unit includes a header, the header is literally inlined into the compilation unit so the more stuff are written in it, the more work it is for the compiler to build the unit (maybe modern C++ compilers have some clever caching mechanism to avoid double compiling this sort of method implementation but in old C times the compiler would parse and build the function every time it sees it, and if you don't make it static then the linker will loudly complain about the multiple definitions). That's why the "best practice" is to only keep prototypes (and class declarations) in the headers and keep the actual implementation (especially large functions definitions) in .cpp files which get compiled orthogonally from one another (and linked later)

In this situation, it's not a big deal but I'm curious if there's a coding style preference that made you do this or it just grew organically into this

Reply:
This one (the `reflash()` function) just grew organically and I simply forgot to move it to the .cpp file.  Done.